crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4269,7 +4269,20 @@ impl<'a> Checker<'a> {
             }
         }
 
-        let scope = self.ctx.scope();
+        // Per [PEP 572](https://peps.python.org/pep-0572/#scope-of-the-target), named
+        // expressions in generators and comprehensions bind to the parent scope.
+        let scope_id = if binding.kind.is_named_expr_assignment() {
```

Comment:
Nit: Would it make sense to return the `scope` instead
```suggestion
        let scope = if binding.kind.is_named_expr_assignment() {
```

It would avoid retrieving the id from the ancestor scope only to look it up again.

Reply:
Yeah, it has to be `mut` though. I'll try to get it working.